src/server/qgsconfigcache.h

```diff
@@ -30,20 +31,59 @@
 #include "qgsproject.h"
 #include "qgsserversettings.h"
 
+#ifndef SIP_RUN
+
+
+#endif
+
+class QgsConfigCache;
+
+/**
+ * \ingroup server
+ * \brief Abstract cache strategy for server configuration
+ * \since QGIS 3.24
+ */
+class SERVER_EXPORT QgsAbstractCacheStrategy  : public QObject
```

Comment:
Why is this a QObject?
Is there any benefit in making this public api? I guess that could be moved into a private header easily.

Reply:
I was concerned about testability, so there should be a way to test strategy changes directly from `QgsConfigCache`. 

If we made these private we need to expose public api to change strategy with different parameters.

I can add method for activating strategy like `setFileSystemStrategy(...)` and make the strategy private and so on but we will loose the generericity of the approach and I'm pretty sure that these methods will stick around for a long time.

Reply:
Why is this a QObject?

Is modifying environment variables not enough for testing purpose?

The effect of making this public is that we will not be able to modify this api any more in the future, so there is a risk this will block some required fixes/improvements in the future. That's why I think we should only make things public that are meant to be public. I.e. if we ever have a registry where someone could register customer implementations of caching strategies, this will need to be public api, but at the current point in time I don't think this is justified.

Reply:
> Is modifying environment variables not enough for testing purpose?

It looks like no:  once you get the instance you won't build able to change the strategy

> The effect of making this public is that we will not be able to modify this api any more in the future

I'm aware of that, may be we can mark that api as experimental ? This is a common practice.

Reply:
I don't see this as experimental, it's an internal implementation detail.

Reply:
Anyway, for testability we need to be able to set the strategy in a unitary way...

Reply:
I would prefer to use the env and completely reinitialize everything.
Otherwise at least kick it out of the python bindings.

Reply:
> I would prefer to use the env and completely reinitialize everything.

How would you do that ? Recreating a `QgServer` is not enough since `init()`, createInstance()`,  rely on all statics variables...

> Otherwise at least kick it out of the python bindings.

Maybe the only viable solution for testing, even if I found that no very convenient (need no reimplement `touch()` :-/)